testplan/testing/base.py

```diff
@@ -529,7 +539,13 @@ def test_command(self):
         :return: Command to run test process
         :rtype: ``list`` of ``str``
         """
-        return [self.cfg.binary]
+        if self.cfg.pre_args:
+            cmd = [*self.cfg.pre_args, self.cfg.binary]
+        else:
+            cmd = [self.cfg.binary]
+        if self.cfg.post_args:
+            cmd.extend(self.cfg.post_args)
+        return cmd
```

Comment:
Similar logic is repeated in derived class, and I think we could avoid this. One possible way is to move the original logic to test_command to _test_command (and inheritances will override accordingly), and in base class test_command, add pre/post_args (and inheritances do not need to override).